### ISSUE TITLE
[Feature Request] let custom adapter method have the same API that CRUD method have

### ISSUE THREAD

**Issue by clarkorz** [balderdashy/waterline#455]
I want to add an `exist`method in my custom adapter, currently, I have to use this method like I use `native()` in `sails-mongo`: No Waterline Query Language supports, no promise supports, the query `options` have different structure, ..., many downsides.

You may ask that why not use `find` or `count` to simulate `exist`. If I my custom adapter works with a database, then you are right. But actually, my custom adapter works with a REST API, `find` and `exist` require to use different endpoints, and there is not `count` or `findOne`.

Hence I suggest, let custom adapter method have the same API that CRUD method have,(except `native()` and `query()` of cause), so we can use model like this if the adapter support an `exist()` method:
 
    Channel.exist()
        .where(actionUtil.parseCriteria(req))
        .then(function(exist) {
          // exist is boolean
        })
        .fail(function(err) {
          // handle err
        })
        .done();

**Comment by dmarcelino** [comment]
@clarkorz, I believe in order to keep custom adapter methods flexible it would be up to the adapter to implement this when possible / desirable. For instance, a custom adapter method may be sync'ed and may not even return anything which would make it really painful for waterline core to do anything about it...

**Comment by sailsbot** [comment]
Thanks for posting, @clarkorz.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!
